server/form-pages/apply/before-you-start/hdc-licence-dates/hdcLicenceDates.ts

```diff
@@ -76,15 +77,33 @@ export default class HDCLicenceDates implements TaskListPage {
 
   errors() {
     const errors: TaskListErrors<this> = {}
-    if (!dateAndTimeInputsAreValidDates(this.body, 'hdcEligibilityDate')) {
+
+    if (dateIsBlank(this.body, 'hdcEligibilityDate')) {
       errors.hdcEligibilityDate = "Enter the applicant's HDC eligibility date"
     }
-    if (!dateAndTimeInputsAreValidDates(this.body, 'conditionalReleaseDate')) {
+    if (dateIsBlank(this.body, 'conditionalReleaseDate')) {
       errors.conditionalReleaseDate = "Enter the applicant's conditional release date"
     }
+
+    if (errors.hdcEligibilityDate || errors.conditionalReleaseDate) {
```

Comment:
do we want to return early at this point? This would mean if one field was blank, but the other had an invalid date, it wouldn't capture it 
![Screenshot 2024-04-24 at 16 48 49](https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/assets/16647486/f051e289-d252-4fbc-b4e7-1533b0d940de)
would it make more sense to group the statements by field rather than condition? Something like
```
    if (dateIsBlank(this.body, 'hdcEligibilityDate')) {
      errors.hdcEligibilityDate = "Enter the applicant's HDC eligibility date"
    } else if (!dateAndTimeInputsAreValidDates(this.body, 'hdcEligibilityDate')) {
      errors.hdcEligibilityDate = 'Eligibility date must be a real date'
    }

    if (dateIsBlank(this.body, 'conditionalReleaseDate')) {
      errors.conditionalReleaseDate = "Enter the applicant's conditional release date"
    } else if (!dateAndTimeInputsAreValidDates(this.body, 'conditionalReleaseDate')) {
      errors.conditionalReleaseDate = 'Conditional release date must be a real date'
    } else if (!dateIsTodayOrInTheFuture(this.body, 'conditionalReleaseDate')) {
      errors.conditionalReleaseDate = 'Conditional release date cannot be in the past'
    }

    if (errors.hdcEligibilityDate || errors.conditionalReleaseDate) {
      return errors
    }
```
Is that really gross to have too many `else if`s I don't know it's the end of the day 😅  but I guess that is what we want to say here? We want to go through certain checks before getting to the next one...